errors.go

```diff
@@ -29,3 +30,54 @@ func New(text string) error {
 func Unwrap(err error) error {
 	return errors.Unwrap(err)
 }
+
+// Last finds the last error in err's chain that matches target, and if one is found, sets
+// target to that error value and returns true. Otherwise, it returns false.
+//
+// The chain consists of err itself followed by the sequence of errors obtained by
+// repeatedly calling Unwrap.
+//
+// An error matches target if the error's concrete value is assignable to the value
+// pointed to by target, or if the error has a method `As(any) bool` such that
+// As(target) returns true.
+//
+// An error type might provide an As() method so it can be treated as if it were a
+// different error type.
+//
+// Last panics if target is not a non-nil pointer to either a type that implements
+// error, or to any interface type.
+//
+// NOTE: Last() is much slower than As(). Therefore As() should always be used
+// unless you absolutely need Last() to retrieve the last error in the error chain
+// that matches the target.
+func Last(err error, target any) bool {
```

Comment:
The "last" and "first" concepts seem ambiguous.  I interpretted "last error" as the last error in chronology.  And the "first" being the first to occur.

Maybe: `errors.Origin()`?

Reply:
Too late.  He merged. 😭